### PR TITLE
remove testleak

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve? 
This is the first step to remove the dependency on TiDB. #677 

### What is changed and how it works?
- remove testleak

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes
N/A

Side effects
N/A

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
-	  GO111MODULE=on go test -p 1 -race -covermode=atomic -coverprofile=coverage.txt -coverpkg=./... ./...
-} || {
-	  mv go.mod go.mod1
-	  mv go.sum go.sum1
-}
+#!/usr/bin/env bash
 
-mv go.mod go.mod1
-mv go.sum go.sum1
+exitCode=0
```
Comment (kennytm) [inline test.sh]: Please just use `set -e`. 

In the current implementation, if `go test` failed the shell script would still `exit 0`.

Comment (tangenta) [reply test.sh]: But the command `mv go.mod go.mod1 && mv go.sum go.sum1` should be executed regardless of the test succeeds or fails.

Comment (kennytm) [reply test.sh]: @tangenta 

```sh
mv go.mod go.mod1 && mv go.sum go.sum1 || true
```

Comment (kennytm) [reply test.sh]: @tangenta You could also use `trap EXIT`, or retain the previous code structure.

**Comment by tiancaiamao** [comment]
I think we can remove the leaktest instead.
The parser doesn't start goroutines. @tangenta

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**